code/modules/hydroponics/plant_genes.dm

```diff
@@ -292,166 +304,222 @@
-			if(C.charge < newcharge)
-				C.charge = newcharge
-				if(isobj(C.loc))
-					var/obj/O = C.loc
-					O.update_icon() //update power meters and such
-				C.update_icon()
-				batteries_recharged = 1
-		if(batteries_recharged)
-			to_chat(target, "<span class='notice'>Your batteries are recharged!</span>")
-
+			target_carbon.electrocute_act(round(power), our_plant, 1, SHOCK_NOGLOVES)
 
+/*
+ * Recharges every cell the person is holding for a bit based on plant potency.
+ *
+ * our_plant - our source plant, that we consumed to charge the cells
+ * eater - the mob that bit the plant
+ * feeder - the mob that feed the eater the plant
+ */
+/datum/plant_gene/trait/cell_charge/proc/recharge_cells(obj/item/our_plant, mob/living/eater, mob/feeder)
+	SIGNAL_HANDLER
 
+	to_chat(eater, "<span class='notice'>You feel energized as you bite into [our_plant].</span>")
+	var/batteries_recharged = FALSE
+	var/obj/item/seeds/our_seed = our_plant.get_plant_seed()
+	for(var/obj/item/stock_parts/cell/found_cell as anything in eater.GetAllContents())
+		var/newcharge = min(our_seed.potency*0.01*found_cell.maxcharge, found_cell.maxcharge)
+		if(found_cell.charge < newcharge)
+			found_cell.charge = newcharge
+			if(isobj(found_cell.loc))
+				var/obj/cell_location = found_cell.loc
+				cell_location.update_icon() //update power meters and such
+			found_cell.update_icon()
+			batteries_recharged = TRUE
+	if(batteries_recharged)
+		to_chat(eater, "<span class='notice'>Your batteries are recharged!</span>")
+
+/// Makes the plant glow. Makes the plant in tray glow, too.
+/// Adds 1 + potency*rate light range and potency*(rate + 0.01) light_power to products.
 /datum/plant_gene/trait/glow
-	// Makes plant glow. Makes plant in tray glow too.
-	// Adds 1 + potency*rate light range and potency*(rate + 0.01) light_power to products.
+
 	name = "Bioluminescence"
 	rate = 0.03
 	examine_line = "<span class='info'>It emits a soft glow.</span>"
 	trait_id = GLOW_ID
 	var/glow_color = "#C3E381"
 
-/datum/plant_gene/trait/glow/proc/glow_range(obj/item/seeds/S)
-	return 1.4 + S.potency*rate
+/datum/plant_gene/trait/glow/proc/glow_range(obj/item/seeds/seed)
+	return 1.4 + seed.potency*rate
 
-/datum/plant_gene/trait/glow/proc/glow_power(obj/item/seeds/S)
-	return max(S.potency*(rate + 0.01), 0.1)
+/datum/plant_gene/trait/glow/proc/glow_power(obj/item/seeds/seed)
+	return max(seed.potency*(rate + 0.01), 0.1)
 
-/datum/plant_gene/trait/glow/on_new(obj/item/food/grown/G, newloc)
-	. = ..()
-	G.light_system = MOVABLE_LIGHT
-	G.AddComponent(/datum/component/overlay_lighting, glow_range(G.seed), glow_power(G.seed), glow_color)
+/datum/plant_gene/trait/glow/on_new(obj/item/our_plant, newloc)
+	if(!..())
+		return
+
+	var/obj/item/seeds/our_seed = our_plant.get_plant_seed()
+	our_plant.light_system = MOVABLE_LIGHT
+	our_plant.AddComponent(/datum/component/overlay_lighting, glow_range(our_seed), glow_power(our_seed), glow_color)
```

Comment:
This could be handled through a setter proc, something like `set_light_system()`
It will have to handle removing the old system if any.

Feel free to disregard this if you are unwilling to do it, as it's out of scope.

Reply:
Not really too familiar with the overlay lighting system so I might leave it for now